course/models.py

```diff
@@ -142,7 +152,42 @@ def create_enrollment_code(sender, instance, created, **kwargs):
         instance.personal_code = code
         instance.save()
 
+def create_anon_id(sender, instance, created, **kwargs):
+    if created or not instance.anon_id:
+        nums = '0123456789'
+        course_id = instance.course_instance.course.code + instance.course_instance.instance_name
+        code = course_id + get_random_string(16, nums)
+        while Enrollment.objects.filter(course_instance=instance.course_instance, anon_id=code).exists():
+            code = course_id + get_random_string(16, nums)
+        instance.anon_id = code
+        instance.save()
+
+def pseudonymize(sender, instance, created, **kwargs):
+    if created or not instance.anon_name:
+        def namegen():
+            num_cols = len(data["colors"])
+            num_ani = len(data["animals"])
+
+            def color():
+                return data["colors"][randint(0, num_cols)]["name"]
+            '''
+             If the color-animal pairs are starting to run out, add another color.
+             This is highly unlikely, as there are roughly 140*68=9520 possible combinations
+            '''
+            second_name = ""
+            if Enrollment.objects.count() > num_cols * num_ani * 0.75:
```

Comment:
`Enrollment.objects.count()` tää sisältänee kaikkien kurssien kaikki enrollaajat, joten varmaa pitäis rajoittaa per kurssi?